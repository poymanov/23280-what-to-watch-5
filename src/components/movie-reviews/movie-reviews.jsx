import React, {PureComponent} from "react";
import MovieTypes from "../../types/movies";
import {connect} from "react-redux";
import {fetchMovieReviews} from "../../store/api-actions";
import PropTypes from "prop-types";
import MovieReview from "../movie-review/movie-review";

class MovieReviews extends PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.fetchMovieReviews(this.props.movie.id);
  }

  render() {
    const {reviews} = this.props;

    if (!reviews) {
      return null;
    }

    return (
      <div className="movie-card__reviews movie-card__row">
        <div className="movie-card__reviews-col">
          {reviews.map((review) => <MovieReview key={review.id} review={review} />)}
        </div>
      </div>
    );
  }
}

MovieReviews.propTypes = {
  movie: MovieTypes.item,
  fetchMovieReviews: PropTypes.func.isRequired,
  reviews: MovieTypes.reviewsList
};

const mapStateToProps = ({MOVIES}) => ({
  reviews: MOVIES.currentMovieReviews,
});

const mapDispatchToProps = (dispatch) => ({
  fetchMovieReviews(movieId) {
    dispatch(fetchMovieReviews(movieId));
  },
});

export {MovieReviews};
export default connect(mapStateToProps, mapDispatchToProps)(MovieReviews);

