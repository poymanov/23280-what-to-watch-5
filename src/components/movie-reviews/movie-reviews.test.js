import React from "react";
import renderer from "react-test-renderer";
import {MovieReviews} from "./movie-reviews";

const noop = () => {};

const movie = {
  id: 1,
  name: `test`,
  posterImage: `test`,
  previewImage: `test`,
  backgroundImage: `test`,
  backgroundColor: `test`,
  videoLink: `test`,
  previewVideoLink: `test`,
  description: `test`,
  rating: 1,
  scoresCount: 1,
  director: `test`,
  starring: [`test`],
  runTime: 1,
  genre: `test`,
  released: 1,
  isFavorite: false,
};

const reviews = [{
  id: 1,
  comment: `test`,
  date: `2020-12-20 12:12:12`,
  rating: 5,
  user: {
    id: 1,
    name: `test`,
  }
}];

describe(`Should MovieReviews render correctly`, () => {
  it(`Without reviews`, () => {
    const tree = renderer
      .create(<MovieReviews fetchMovieReviews={noop} movie={movie} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With reviews`, () => {
    const tree = renderer
      .create(<MovieReviews fetchMovieReviews={noop} movie={movie} reviews={reviews} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
