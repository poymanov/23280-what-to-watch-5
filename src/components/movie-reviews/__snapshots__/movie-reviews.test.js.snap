// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MovieReviews render correctly With reviews 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          test
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            test
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            December 20, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        5
      </div>
    </div>
  </div>
</div>
`;

exports[`Should MovieReviews render correctly Without reviews 1`] = `null`;
